source/colors.py

```diff
@@ -1,24 +1,62 @@
-#colors.py
-#A part of NonVisual Desktop Access (NVDA)
-#Copyright (C) 2006-2008 NVDA Contributors <http://www.nvda-project.org/>
-#This file is covered by the GNU General Public License.
-#See the file COPYING for more details.
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2006-2021 NV Access Limited, Babbage B.V., Joseph Lee, Mesar Hameed, kvark128
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
 
 from collections import namedtuple
 import math
 import colorsys
 from ctypes.wintypes import COLORREF
 import re
+from functools import lru_cache
+from typing import Union
+
+#: Flag to indicate color being decoded from displayModelFormatColor_t
+# is transparent.
+# See (displayModel.cpp in nvdaHelper) displayModelFormatColor_t::TRANSPARENT_BIT
+TRANSPARENT_BITFLAG = 0x01 << 24
 
 class RGB(namedtuple('RGB',('red','green','blue'))):
 	"""Represents a color as an RGB (red green blue) value"""
 
+	#: The transparency of the color.
+	# Note: may be useful for add-ons or app modules who wish to customize color reporting
+	# of background colors of applications that rely on the display model.
+	# These applications do not always reliably report their background color.
+	# The background color may be reported for text with transparent backgrounds.
+	# In other cases the background for the text is transparent, but the background color is
+	# visually correct and rendered in a different way EG via GDI filledRect.
+	alphaValue: int = 0xFF  # no transparency by default
+
+	@classmethod
+	def fromDisplayModelFormatColor_t(cls, c: int) -> "RGB":
+		"""factory method to create an RGB from a DisplayModelFormatColor_t
+			Color format is 4 bytes:  0xTTbbggrr
+			TT bit flags, only bit 1 used: set for transparent.
+			Encoding alpha in TT was considered (eg 0xFFbbggrr for opaque),
+			but this would break compatibility with code that continues to pass
+			0x00bbggrr for opaque, all usages would need to be fixed.
+		"""
+		rr = c & 0xFF
+		gg = (c >> 8) & 0xFF
+		bb = (c >> 16) & 0xFF
+		tt = c & TRANSPARENT_BITFLAG
+		rgb = cls(rr, gg, bb)
+		rgb.alphaValue = 0x00 if bool(tt) else 0xFF

```

Comment:
Could we use constants for this instead?

Reply:
Yep, good point.